### ISSUE TITLE
Kubelet - add metrics related to images pull time

### ISSUE THREAD

**Issue by grzesuav** [kubernetes/kubernetes#111259]
### What would you like to be added?

It would be helpful to see metrics around pulling an image by kubelet (related code around [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/images/image_manager.go#L144) ), i.e.:
* number of currently pulling images - gauge
* time of pulling an image (maybe histogram) ?
* total errors related to pulling - counter ?

### Why is this needed?

It would allow to observe and alert issues related with images pulling

**Comment by pacoxu** [comment]
It makes sense. 
Do you plan to work on this? If not, I may assign it.
It would be helpful for users to move to use `parallel image pulls` like https://github.com/kubernetes/kubernetes/issues/108405.

**Comment by DrAuYueng** [comment]
Let me try.

**Comment by DrAuYueng** [comment]
Let me try, same needs.

**Comment by pacoxu** [comment]
https://github.com/kubernetes/kubernetes/issues/111704 inspired that the pulling time of images were already included in the image pulled event like 
```
0s          Normal   Pulled      pod/pod-small-3   Successfully pulled image "docker.m.daocloud.io/library/ubuntu:22.10" in 49.18401355s
```

**Comment by pacoxu** [comment]
I tried to add metrics in containerd. https://github.com/containerd/containerd/pull/7313
Would you take a look?

**Comment by pacoxu** [comment]
https://github.com/containerd/containerd/pull/7313 is merged. 
And a thread is started for parallel image pulling in https://github.com/kubernetes/enhancements/issues/3673.

**Comment by ialidzhikov** [comment]
https://github.com/kubernetes/enhancements/issues/2371#issuecomment-1961342677: I raised the question how the containerd metrics for image pull should be consumed - via scrapping containerd directly or will be they exposed somehow via kubelet?

**Comment by sidewinder12s** [comment]
/remove-lifecycle rotten

Pending confirmation this is fixed in 1.30+

**Comment by sidewinder12s** [comment]
/remove-lifecycle rotten

Pending confirmation this is fixed in 1.30+, I hadn't upgraded yet though will be this quarter.

**Comment by SergeyKanzhelev** [comment]
/close

please reopen if there is a bug. Closing for now
